src/main/java/cam72cam/immersiverailroading/physics/PhysicsThread.java

```diff
@@ -0,0 +1,63 @@
+package cam72cam.immersiverailroading.physics;
+
+import cam72cam.immersiverailroading.Config.ConfigPerformance;
+import cam72cam.immersiverailroading.ImmersiveRailroading;
+import cam72cam.mod.world.World;
+
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+/**
+ * Utility class for the physics thread.
+ */
+public class PhysicsThread {
+    /**
+     * The active physics thread, or null if none is active.
+     */
+    @Nullable
+    private static ExecutorService executor;
+
+    /**
+     * The last job submitted to the physics thread by dimension id.
+     */
+    private static final HashMap<Integer, Future<?>> lastJobs = new HashMap<Integer, Future<?>>();
+
+    /**
+     * Gets or creates the physics thread, if enabled, or null if disabled.
+     * @return The physics thread, or null if disabled.
+     */
+    @Nullable
+    private static ExecutorService getExecutor() {
+        if (ConfigPerformance.multithreadedPhysics) {
+            if (executor == null) {
+                ImmersiveRailroading.info("Starting physics thread");
+                executor = Executors.newSingleThreadExecutor((t) -> new Thread(t, "ImmersiveRailroading Physics Thread"));
```

Comment:
This could now could have multiple threads

Reply:
Maybe that could be ConfigPerformance.multithreadedPhysics, where any value greater than 0 is the number of threads to use